File: src/main/java/venus/model/UserCourse.java
package venus.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name = "users_courses")
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class UserCourse extends BaseEntity {

    @ManyToOne
    @JoinColumn(name = "userid")
    private User student;

    @ManyToOne
    @JoinColumn(name = "courseid")
    private Course course;

    @Column
    private int mark;
}

File: frontend/src/app/model/user-course.ts
export class UserCourse {
    userId: number;
    username: string;
    email: string;
    mark: number;

    constructor(
        userId?: number,
        username?: string,
        email?: string,
        mark?: number) {

        this.userId = userId || 0;
        this.username = username || "";
        this.email = email || "";
        this.mark = mark || 0;
    }
}